RHEL6_7/storage/dasd-format/check

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+. /usr/share/preupgrade/common.sh
+#END GENERATED SECTION
+
+ARCH="$(arch)"
+
+DASDVIEW=$(which dasdview)
+DASDVIEW=${DASDVIEW:-/sbin/dasdview}
+
+GREP=$(which grep)
+GREP=${GREP:-/bin/grep}
+
+STR_IFS=$IFS
+IFS=" $(echo -n -e '\t')"
+
+function wrap_exit
+{
+	IFS=$STR_IFS
+
+	exit $1
+}
+
+[ "$ARCH" == "s390x" ] || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $DASDVIEW || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $GREP || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $CUT || wrap_exit $RESULT_NOT_APPLICABLE
+
+
+function detect_linux_disk_layout()
+{
+	log_debug "Checking Linux Disk Layout for a device $1"
+
+	local format
+	format=$($DASDVIEW -x $1 2>/dev/null | $GREP "format :" | $CUT -f 3)
```

Comment:
Not sure that subcommand is correct. Probably there is used `\t` as a separator but in that case, regexp is incorrect and cut as well (in case regexp would be correct, it would return just `hex` substring). Example of the line (probably tabulators have been replaced by spaces already:
```
format                 : hex 2          dec 2           CDL formatted
```

In such case, the output would be not catched. From the code itself, I guess you want to get directly the `LDL`/`CDL` (`-f 7`) substring or everything after hex (`-f 3-`). Anyway, it seems simpler to me just get the line and then grep it for LDL or CDL substring (comment below).
```suggestion
	format=$($DASDVIEW -x $1 2>/dev/null | $GREP "^[:spaces:]*format[:spaces:]*:")
```